src/azure-cli-core/azure/cli/core/extension/tests/latest/test_extension_commands.py

```diff
@@ -387,6 +388,45 @@ def test_update_extension_extra_index_url(self):
         ext = show_extension(MY_EXT_NAME)
         self.assertEqual(ext[OUT_KEY_VERSION], '0.0.4+dev')
 
+    def test_add_extension_to_path(self):
+        add_extension(cmd=self.cmd, source=MY_EXT_SOURCE)
+        num_exts = len(list_extensions())
+        self.assertEqual(num_exts, 1)
+        ext = get_extension('myfirstcliextension')
+        old_path = sys.path[:]
+        try:
+            add_extension_to_path(ext.name)
+            self.assertSequenceEqual(old_path, sys.path[:-1])
+            self.assertEqual(ext.path, sys.path[-1])
+        finally:
+            sys.path[:] = old_path
+
+    def test_add_extension_azure_to_path(self):
+        import azure
+        import azure.mgmt
+        old_path_1 = azure.__path__
+        old_path_2 = azure.mgmt.__path__
```

Comment:
@zooba The CI log shows `azure.mgmt.__path__` is an object of `_NamespacePath`. The old path will have the new value when the object gets updated.
```suggestion
        old_path_1 = list(azure.__path__)
        old_path_2 = list(azure.mgmt.__path__)
```

Reply:
I used pytest to test it directly, the issue does not exist and debug shows `azure.mgmt.__path__` is a list. CI uses [tox](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/tox.ini) and it fails.

Reply:
It's going to depend on whether you've used the `pkg_resources` call in your `__init__.py` files. You really shouldn't need those anywhere, though you do need to be careful about not using old versions of the tools, because some of those still assume they're necessary.

You also need to be careful about not putting an `__init__.py` _anywhere_ during development, as that will prevent you importing the other copies elsewhere on `sys.path`. Or you can use the same `__path__` trick to make local development easier. (Installs with pip use the `nspkg` packages to ensure the final directory structure is correct, and installs with setuptools are not supported.)

Reply:
We do use `pkg_resources` now and we're planning to follow PEP 420 to remove it. This does not affect the `__path__` trick you add, right? If so, let's fix the CI tests to unblock this PR first.

Reply:
Linking to https://github.com/Azure/azure-cli/issues/13293 to following PEP 420

Reply:
The `__path__` trick is only necessary with the `nspkg` packages, which I expect you'll keep to avoid the hijacking issues I mentioned below.

If you remove those, then yeah, extensions can just use the normal hijacking approach to extend the `azure` namespace. So no need for this change.

Reply:
I was wrong above - it's a `_NamespacePath` _because of_ PEP 420. So because it's no longer a real list, we have to remove the extension and trigger an update. Hopefully that will restore the right value.